android-sample-java/app/src/main/java/io/logto/demo4j/viewmodel/LogtoViewModel.java

```diff
@@ -0,0 +1,97 @@
+package io.logto.demo4j.viewmodel;
+
+import android.app.Activity;
+import android.app.Application;
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import io.logto.sdk.android.LogtoClient;
+import io.logto.sdk.android.exception.LogtoException;
+import io.logto.sdk.android.type.AccessToken;
+import io.logto.sdk.android.type.LogtoConfig;
+import io.logto.sdk.core.type.IdTokenClaims;
+
+public class LogtoViewModel extends AndroidViewModel {
+
+    public LogtoViewModel(@NonNull Application application) {
+        super(application);
+    }
+
+    private LogtoConfig logtoConfig = new LogtoConfig(
+            "https://logto.dev",
+            "z4skkM1Z8LLVSl1JCmVZO",
+            null,
+            null,
+            true
+    );
+
+    private final LogtoClient logtoClient = new LogtoClient(logtoConfig, getApplication());
+
+    private final MutableLiveData<Boolean> _authenticated = new MutableLiveData<>(logtoClient.isAuthenticated());
+    public LiveData<Boolean> isAuthenticated() {
+        return _authenticated;
+    }
+
+    private final MutableLiveData<AccessToken> _accessToken = new MutableLiveData<>();
+    public LiveData<AccessToken> getAccessToken() {
+        return _accessToken;
+    }
+
+    private final MutableLiveData<IdTokenClaims> _idTokenClaims = new MutableLiveData<>();
+    public LiveData<IdTokenClaims> getIdTokenClaims() {
+        return _idTokenClaims;
+    }
+
+    private final MutableLiveData<LogtoException> _logtoException = new MutableLiveData<>();
+    public LiveData<LogtoException> getLogtoException() {
+        return _logtoException;
+    }
```

Comment:
what's these for?

Reply:
[Android ViewModel](https://developer.android.com/topic/libraries/architecture/viewmodel?gclid=Cj0KCQiAmKiQBhClARIsAKtSj-k1oJZwWN7baeAtovZ4BJzuUOdYT80F7-mXX_eMbkM-leNSlcUtDqkaAqAUEALw_wcB&gclsrc=aw.ds)

Reply:
looks a little bit weird. would you like to add an issue for a deeper dive on this? fine to keep as it is now

Reply:
LOG-1555